modin/numpy/arr.py

```diff
@@ -704,50 +741,99 @@ def mean(self, axis=None, dtype=None, out=None, keepdims=None, *, where=True):
             if dtype is not None
             else (out.dtype if out is not None else self.dtype)
         )
+        out_type = out_dtype if not hasattr(out_dtype, "type") else out_dtype.type
+        if isinstance(where, array) and issubclass(out_type, numpy.integer):
+            out_dtype = numpy.float64
```

Comment:
why?..

Reply:
if we try and convert out to the out dtype on ubuntu, it errors, since out will contain NaNs.

Reply:
As far as I can see, this handles the case when after `where.where()` we have some `NaN`-s inside... but it's not always going to happen? What does numpy do in this case, does it convert the output to `float` regardless of whether `NaN`-s are present?

P.S. I don't understand what "ubuntu" means here 🙃 so I "played by ear" and guessed what you meant, hopefully correctly...

Reply:
Sorry - to clarify, I meant that if we don't do this, our tests fail on the Ubuntu OS (but not on macOS or windows).

Interesting - I just checked `NumPy`, and it looks like if `NaN` is present in the array, it returns `NaN` as the mean (or a junk value if the output dtype is int). We ignore the `NaN` and compute the mean or sum as if it wasn't there - because that's how pandas does it, but NumPy propagates the NaN.

Do you think we should try and resolve this in this PR? I'm concerned its already kind of big, and if we do decide to fix this, this will also mean we need to handle our masks differently, since we currently take advantage of pandas ignoring `NaN` behavior to replace values we're making with `NaN` without affecting the sum, so we'd probably need to come up with a new method for masking. (Although we could technically convert this to a Binary algebra operator maybe? Where the second operand is the where array for masking?)

Reply:
I would say, let's make a separate issue, as this PR is really already big enough 🙃

Reply:
Issue created: https://github.com/modin-project/modin/issues/5720